nmdc/messages.go

```diff
@@ -630,8 +630,7 @@ func (m *MyInfo) unmarshalTag(tag []byte) error {
 		}
 		if i < 0 {
 			return fmt.Errorf("unknown field in tag: %q", field)
-		}
-		if i >= 0 {
+		} else {
```

Comment:
It's better to just omit else and lower the code. Since there is a `return` in a branch above it will work the same.